pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -187,6 +199,33 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 	// negative conditions (!Available, Degraded, Progressing)
 	r.conditions = nil
 
+	// Handle finalizers
+	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
+		// Add the finalizer if it's not there
+		if !contains(instance.ObjectMeta.Finalizers, FinalizerName) {
+			instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, FinalizerName)
+			// Need to requeue because finalizer update does not change metadata.generation
+			return reconcile.Result{}, r.client.Update(context.TODO(), instance)
+		}
+	} else {
+		for _, f := range []func(c client.Client, cr *hcov1alpha1.HyperConverged) error{
```

Comment:
I think there should be another check for finalizer existence in case something went wrong and the finalizer is no set.

Reply:
seems reasonable.

Reply:
There is some good information in the operator-sdk documentation regarding [finalizers](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#handle-cleanup-on-deletion).

Checking for the finalizer and not removing the finalizer unless the cleanup proceeds successfully sounds like a good idea.